RiotSwiftUI/Modules/Authentication/VerifyEmail/Coordinator/AuthenticationVerifyEmailCoordinator.swift

```diff
@@ -199,4 +212,109 @@ final class AuthenticationVerifyEmailCoordinator: Coordinator, Presentable {
         
         authenticationVerifyEmailViewModel.displayError(.unknown)
     }
+    
+    // Tchap: Add account creation part in this class
+    /// Creates an account on the homeserver with the supplied password.
+    @MainActor private func createAccount(password: String) {
+        let authService = AuthenticationService.shared
+        let registrationWizard = authService.registrationWizard
+        guard let registrationWizard = registrationWizard else {
+            MXLog.failure("[AuthenticationVerifyEmailCoordinator] createAccount: The registration wizard is nil.")
+            return
+        }
+        
+        startLoading()
+        
+        currentTask = Task { [weak self] in
+            do {
+                let result = try await registrationWizard.createAccount(username: nil,
+                                                                        password: password,
+                                                                        initialDeviceDisplayName: UIDevice.current.initialDisplayName)
+                
+                guard !Task.isCancelled else { return }
+                
+                switch result {
+                case .flowResponse(let flowResult):
+                    if flowResult.missingStages.contains(.email(isMandatory: true)) {
+                        authenticationVerifyEmailViewModel.context.send(viewAction: .send)
+                    } else {
+                        // Should not happen.
+                        MXLog.error("[AuthenticationVerifyEmailCoordinator] createAccount flowResponse with no e-mail missing stage !")
+                    }
+                case .success:
+                    MXLog.debug("[AuthenticationVerifyEmailCoordinator] createAccount success")
+                    self?.callback?(.completed(result))
+                }
+                
+                self?.stopLoading()
+            } catch {
+                self?.stopLoading()
+                self?.handleRegistrationError(error)
+            }
+        }
+    }
+    
+    /// Processes an error to either update the flow or display it to the user.
+    @MainActor private func handleRegistrationError(_ error: Error) {
```

Comment:
why don't you use the existing handleError(_ error: Error) method?
I think the code of this new method can be added to the existing one

Reply:
It's a mistake after taking part of the content from another class. Fixed in latest commits